storage/range_tree.go

```diff
@@ -73,8 +63,25 @@ func SetupRangeTree(batch engine.Engine, ms *engine.MVCCStats, timestamp proto.T
 	return nil
 }
 
-// GetRangeTree fetches the RangeTree proto.
-func getRangeTree(db *client.KV) (*rangeTree, error) {
+// flush writes all updated nodes and the tree to the transaction.
+func (t *treeContext) flush() error {
+	if t.treeUpdated {
```

Comment:
Actually, the transaction will need to be run around the call to Insert(), as all reads must be transactional too. And keep in mind that the node cache should be cleared each time through the transaction (in case of restarts).